pkg/certificate/manager.go

```diff
@@ -0,0 +1,146 @@
+package certificate
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/k8s/events"
+	"github.com/openservicemesh/osm/pkg/messaging"
+)
+
+var errNoIssuingCA = errors.New("no issuing CA")
+var errCertNotFound = errors.New("certificate not found")
+
+// NewManager creates a new CertManager with the passed CA and CA Private Key
+func NewManager(
+	ca *Certificate,
+	client client,
+	serviceCertValidityDuration time.Duration,
+	msgBroker *messaging.Broker) (*manager, error) {
+	if ca == nil {
+		return nil, errNoIssuingCA
+	}
+
+	m := &manager{
+		// The root certificate signing all newly issued certificates
+		ca:                          ca,
+		client:                      client,
+		serviceCertValidityDuration: serviceCertValidityDuration,
+		msgBroker:                   msgBroker,
+	}
+
+	// TODO(steeling) start the cert rotation here.
+
+	return m, nil
+}
+
+func (m *manager) deleteFromCache(cn CommonName) {
+	m.cache.Delete(cn)
+}
```

Comment:
Why is this wrapper necessary? Because this is un-exported, *manager implementation can use `m.cache.Delete()` directly.

Reply:
FYI all of these function were just copy-pasted, I didn't write anything new in this file. Happy to make the changes though

Reply:
No problem, feel free to update it later if you agree.